File: FivePointTen.java
import java.util.Scanner;


public class FivePointEight {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter employee name: ");
        String name = sc.next();

        System.out.println("Enter employee salary: ");
        double hourly = sc.nextDouble();

        System.out.println("How many hours did " + name + " work? (Input fractional hours as a decimal)");
        double hours = sc.nextDouble();

        Paycheck check = new Paycheck(hours, hourly);
        double pay = check.computePay();

        System.out.println(name + " is owed $" + pay);
    }
}

File: everything_but_in_c/5.10/FivePointTen.c
#include <stdio.h>
#include <stdlib.h>
#include "main.h"

int main(int argc, char** argv) {
    Employee e;
    printf("Enter name of Employee: ");
    scanf("%s", e.name);

    printf("\nEnter salary of %s: ", e.name);
    scanf("%f", e.salary);

    printf("\nEnter number of hours %s worked: ", e.name);
    scanf("%f", e.hours);

    printf("%s is owed $%.2f", e.name, computePay(&e));
}

double computePay(Employee *e) {
    if (e->hours > 40) {
        return (40 * e->salary) + ((e->hours - 40) * (e->salary * 1.5));
    } else {
        return e->hours * e->salary;
    }
}
